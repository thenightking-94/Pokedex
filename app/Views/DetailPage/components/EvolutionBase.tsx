import dynamic from 'next/dynamic';
import React, { ReactElement, useEffect, useState } from 'react';
import { getEvolutionArray } from '../../../Utils/Utils';
import PointerIcon from '../../Icons-svg/pointerIcon';
const PokemonCards = dynamic(import('../../PokemonListing/components/PokemonCards')) ;

export default function EvolutionBase ({ evolutionData}): ReactElement{

       const [evolutionArray, setEvolutionArray] = useState<Array<{name: string, url: string}>>([])

       useEffect(() => {
        if(!!evolutionData && !!evolutionData.chain){
            let evolutionSeries  = getEvolutionArray(evolutionData?.chain, []) 
            setEvolutionArray(evolutionSeries)
        }
       },[evolutionData])


    return(
        <>
        <div className='title'>{'Evolution chain'}</div>
        <div className='pokemon-cards-detail'>
            {
               !!evolutionArray &&  evolutionArray?.length > 0 && [...new Set(evolutionArray)].map((item, index) => {
                    const propsForEach = {item, inDetailPage: true}
                    return (
                        <>
                            <PokemonCards {...propsForEach}/>
                            {!(index === evolutionArray.length-1) &&
                                <div className='pointer'>
                                    <PointerIcon/>
                                </div>
                            }
                        </>
                        
                    )
                })
            }
        </div>
            
        </>
    )

}