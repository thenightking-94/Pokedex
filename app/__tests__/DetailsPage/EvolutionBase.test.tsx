import { render } from '@testing-library/react'
import { QueryClient, QueryClientProvider } from 'react-query'
import renderer from "react-test-renderer";
import EvolutionBase from '../../Views/DetailPage/components/EvolutionBase'

const queryClient = new QueryClient()

const props = {
   evolutionData: {
        "baby_trigger_item": null,
        "chain": {
        "evolution_details": [],
        "evolves_to": [
        {
        "evolution_details": [
        {
        "gender": null,
        "held_item": null,
        "item": null,
        "known_move": null,
        "known_move_type": null,
        "location": null,
        "min_affection": null,
        "min_beauty": null,
        "min_happiness": null,
        "min_level": 16,
        "needs_overworld_rain": false,
        "party_species": null,
        "party_type": null,
        "relative_physical_stats": null,
        "time_of_day": "",
        "trade_species": null,
        "trigger": {
        "name": "level-up",
        "url": "https://pokeapi.co/api/v2/evolution-trigger/1/"
        },
        "turn_upside_down": false
        }
        ],
        "evolves_to": [
        {
        "evolution_details": [
        {
        "gender": null,
        "held_item": null,
        "item": null,
        "known_move": null,
        "known_move_type": null,
        "location": null,
        "min_affection": null,
        "min_beauty": null,
        "min_happiness": null,
        "min_level": 36,
        "needs_overworld_rain": false,
        "party_species": null,
        "party_type": null,
        "relative_physical_stats": null,
        "time_of_day": "",
        "trade_species": null,
        "trigger": {
        "name": "level-up",
        "url": "https://pokeapi.co/api/v2/evolution-trigger/1/"
        },
        "turn_upside_down": false
        }
        ],
        "evolves_to": [],
        "is_baby": false,
        "species": {
        "name": "charizard",
        "url": "https://pokeapi.co/api/v2/pokemon-species/6/"
        }
        }
        ],
        "is_baby": false,
        "species": {
        "name": "charmeleon",
        "url": "https://pokeapi.co/api/v2/pokemon-species/5/"
        }
        }
        ],
        "is_baby": false,
        "species": {
        "name": "charmander",
        "url": "https://pokeapi.co/api/v2/pokemon-species/4/"
        }
        },
        "id": 2
        }
}


describe('Evolution', () => {
  it('renders correctly', () => {
    render(
    <QueryClientProvider client={queryClient}>
         <EvolutionBase {...props} />
    </QueryClientProvider>
   )
  })
})

describe("Jest Snapshot testing suite", () => {
    it("Matches DOM Snapshot", () => {
      const domTree = renderer.create( 
      <QueryClientProvider client={queryClient}>
        <EvolutionBase {...props} />
     </QueryClientProvider>).toJSON();
      expect(domTree).toMatchSnapshot();
    });
  });