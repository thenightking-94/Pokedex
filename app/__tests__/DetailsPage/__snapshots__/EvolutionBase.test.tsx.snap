// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite Matches DOM Snapshot 1`] = `
[
  <div
    className="title"
  >
    Evolution chain
  </div>,
  <div
    className="pokemon-cards-detail"
  />,
]
`;
